library/ajax/weno_pharmacy_search.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+/**
+ * weno pharmacy search.
+ *
+ * @package   OpenEMR
+ * @link      https://www.open-emr.org
+ * @author    Kofi Appiah <kkappiah@medsov.com>
+ * @copyright Copyright (c) 2023 Kofi Appiah <kkappiah@medsov.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+ require_once(dirname(__FILE__) . "/../../interface/globals.php");
+
+if (isset($_GET['searchFor']) && $_GET['searchFor'] == 'weno_city') {
+    $return_arr = array();
+    $term    = filter_input(INPUT_GET, "term");
+
+    $sql = "SELECT city, id FROM weno_pharmacy WHERE city LIKE ? LIMIT 10";
+    $val = array('%' . $term . '%');
+    $res = sqlStatement($sql, $val, $val);
+    while ($row = sqlFetchArray($res)) {
+        $return_arr[] =  $row['city'];
+    }
+
+    echo json_encode($return_arr);
+}
+
+if (isset($_GET['searchFor']) && $_GET['searchFor'] == 'weno_pharmacy') {
+    $term    = filter_input(INPUT_GET, "term");
+    $val = '%' . $term . '%';
+
+    $sql = "SELECT business_name, state, ncpdp, city, address_line_1 " .
+            "FROM weno_pharmacy WHERE Business_Name LIKE '$val'";
```

Comment:
placing variables directly into a sql query will create a sql injection security vulneerability. Need to use placeholders (ie. `?`) instead and then pass the variables as an array for the second parameter when run the query. In cases like this where building dynamic number of parameters while building the query, recommend creating a $sqlArray variable that will be an array and then add your variables to it whil building the query (and use `?` placeholder in the query). Then when run the query use the $sqlArray as secondary parameter.

Reply:
Great, will do